### DISCUSSION TITLE
Use MRKL with other bots/chains

### DISCUSSION THREAD

**Discussion by Captain-Bacon** [FlowiseAI/Flowise#33] [General]
I don't think it's possible at the moment to have a MRKL module interact with a pinecone embedding unless you have set the url to scrape or the github repo to search (for example) at the outset.

For example I wanted to get it to read a github repo, have that knowledge all squared away in a tight little 512 dimension vector and have the MRKL module be able to access the QnA retriever as an agent. BUT I want it to decide that it needs to do that itself, so have the github repo url as an value that can be taken from the MRKL agent.

Any thoughts on that, or am i being blind or stupid?

**Comment by HenryHengZJ** [comment]
Not sure if this is what you're looking for but giving a shot.

In Flowise, there is a template called: `Multiple Vector Stores`. Assuming you already have the knowledge/doc upserted to vector databases, you can have the agent automatically decide which one to use. 

Say if your question/query is more related to a specific topic, the agent will be like:  "i've seen this before, i have to get all the similar searches from this particular database". 

However if you have say multiple github repo/docs upserted into same database, and you want to let the agent to choose which one to search, techinically you'll have to use metadata to filter, but this is currently not supported in LangChainJS

![image](https://user-images.githubusercontent.com/26460777/234414681-bce39735-abd1-45e5-b585-ec4431ced7f6.png)
